include/re_main.h

```diff
@@ -35,10 +36,10 @@ typedef void (fd_h)(int flags, void *arg);
 typedef void (re_signal_h)(int sig);
 
 
-int   fd_listen(re_sock_t fd, int flags, fd_h *fh, void *arg);
-void  fd_close(re_sock_t fd);
+int   fd_listen(struct re_fhs **fhs, re_sock_t fd, int flags, fd_h *fh,
+		void *arg);
+void  *fd_close(struct re_fhs *fhs);
```

Comment:
Why not return `re_fhs *` here? In C++, you have to explicitly cast `void *` to `re_fhs *` if you want to follow the suggested usage pattern of `p = fd_close(p)`.